docs/monorepo/history_strategy.md

```diff
@@ -0,0 +1,218 @@
+# La Grande Fusion with History
```

Comment:
`La Grade Fusion` is a great code name, but probably in docs that are going to live for a while, we should use more boring terms =) Maybe: "Retaining history when merging the engine and framework repos".